tests: IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR (PHP 8.2.12)

The automatic test suite started failing for PHP 8.1 and 8.2 after https://github.com/shaarli/Shaarli/pull/2028 which is just a documentation update and doesn't touch dependencies or application code.

https://github.com/shaarli/Shaarli/actions/workflows/ci.yml

```
-------
PHPUNIT
-------
PHPUnit 9.6.13 by Sebastian Bergmann and contributors.

Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

...............................................................  63 / 993 (  6%)
............................................................... 126 / 993 ( 12%)
............................................................... 189 / 993 ( 19%)
............................................................... 252 / 993 ( 25%)
............................................................... 315 / 993 ( 31%)
............................................................... 378 / 993 ( 38%)
.............................E................................. 441 / 993 ( 44%)
...........................................................EEEE 504 / 993 ( 50%)
E..E.E.........E............................................... 567 / 993 ( 57%)
............................................................... 630 / 993 ( 63%)
.EEEE.EEEE..................................................... 693 / 993 ( 69%)
............................................................... 756 / 993 ( 76%)
............................................................... 819 / 993 ( 82%)
............................................................... 882 / 993 ( 88%)
............................................................... 945 / 993 ( 95%)
................................................                993 / 993 (100%)

Time: 00:08.695, Memory: 62.00 MB

There were 17 errors:

1) Shaarli\Front\Controller\Admin\ServerControllerTest::testIndex
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/front/controller/admin/ServerController.php:48
/home/runner/work/Shaarli/Shaarli/tests/front/controller/admin/ServerControllerTest.php:61
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

2) Shaarli\Front\Controller\Visitor\DailyControllerTest::testValidIndexControllerInvokeDefault
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/application/front/controller/visitor/DailyController.php:35
/home/runner/work/Shaarli/Shaarli/tests/front/controller/visitor/DailyControllerTest.php:100
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

3) Shaarli\Front\Controller\Visitor\DailyControllerTest::testValidIndexControllerInvokeNoFutureOrPast
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/application/front/controller/visitor/DailyController.php:35
/home/runner/work/Shaarli/Shaarli/tests/front/controller/visitor/DailyControllerTest.php:223
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

4) Shaarli\Front\Controller\Visitor\DailyControllerTest::testValidIndexControllerInvokeHeightAdjustment
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/application/front/controller/visitor/DailyController.php:35
/home/runner/work/Shaarli/Shaarli/tests/front/controller/visitor/DailyControllerTest.php:281
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

5) Shaarli\Front\Controller\Visitor\DailyControllerTest::testValidIndexControllerInvokeNoBookmark
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/application/front/controller/visitor/DailyController.php:35
/home/runner/work/Shaarli/Shaarli/tests/front/controller/visitor/DailyControllerTest.php:328
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

6) Shaarli\Front\Controller\Visitor\DailyControllerTest::testValidRssControllerInvokeDefault
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/application/front/controller/visitor/DailyController.php:135
/home/runner/work/Shaarli/Shaarli/tests/front/controller/visitor/DailyControllerTest.php:377
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

7) Shaarli\Front\Controller\Visitor\DailyControllerTest::testSimpleIndexWeekly
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:184
/home/runner/work/Shaarli/Shaarli/application/front/controller/visitor/DailyController.php:35
/home/runner/work/Shaarli/Shaarli/tests/front/controller/visitor/DailyControllerTest.php:522
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

8) Shaarli\Front\Controller\Visitor\DailyControllerTest::testSimpleRssWeekly
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:184
/home/runner/work/Shaarli/Shaarli/application/front/controller/visitor/DailyController.php:135
/home/runner/work/Shaarli/Shaarli/tests/front/controller/visitor/DailyControllerTest.php:634
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

9) Shaarli\Front\Controller\Visitor\InstallControllerTest::testInstallIndexWithValidSession
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/front/controller/visitor/InstallController.php:65
/home/runner/work/Shaarli/Shaarli/tests/front/controller/visitor/InstallControllerTest.php:66
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

10) Shaarli\Helper\DailyPageHelperTest::testGeDescriptionsByType with data set #0 ('day', DateTimeImmutable Object (...), 'Today - November 1, 2023')
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/tests/helper/DailyPageHelperTest.php:122
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

11) Shaarli\Helper\DailyPageHelperTest::testGeDescriptionsByType with data set #1 ('day', DateTimeImmutable Object (...), 'Yesterday - October 31, 2023')
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/tests/helper/DailyPageHelperTest.php:122
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

12) Shaarli\Helper\DailyPageHelperTest::testGeDescriptionsByType with data set #2 ('day', DateTimeImmutable Object (...), 'October 9, 2020')
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/tests/helper/DailyPageHelperTest.php:122
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

13) Shaarli\Helper\DailyPageHelperTest::testGeDescriptionsByType with data set #3 ('week', DateTimeImmutable Object (...), 'Week 41 (October 5, 2020)')
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:184
/home/runner/work/Shaarli/Shaarli/tests/helper/DailyPageHelperTest.php:122
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

14) Shaarli\Helper\DailyPageHelperTest::testGeDescriptionsByTypeNotIncludeRelative with data set #0 ('day', DateTimeImmutable Object (...), 'November 1, 2023')
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/tests/helper/DailyPageHelperTest.php:135
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

15) Shaarli\Helper\DailyPageHelperTest::testGeDescriptionsByTypeNotIncludeRelative with data set #1 ('day', DateTimeImmutable Object (...), 'October 31, 2023')
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/tests/helper/DailyPageHelperTest.php:135
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

16) Shaarli\Helper\DailyPageHelperTest::testGeDescriptionsByTypeNotIncludeRelative with data set #2 ('day', DateTimeImmutable Object (...), 'October 9, 2020')
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:192
/home/runner/work/Shaarli/Shaarli/tests/helper/DailyPageHelperTest.php:135
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

17) Shaarli\Helper\DailyPageHelperTest::testGeDescriptionsByTypeNotIncludeRelative with data set #3 ('week', DateTimeImmutable Object (...), 'Week 41 (October 5, 2020)')
IntlException: datefmt_create: invalid locale: U_ILLEGAL_ARGUMENT_ERROR

/home/runner/work/Shaarli/Shaarli/application/Utils.php:327
/home/runner/work/Shaarli/Shaarli/application/helper/DailyPageHelper.php:184
/home/runner/work/Shaarli/Shaarli/tests/helper/DailyPageHelperTest.php:135
/home/runner/work/Shaarli/Shaarli/vendor/bin/phpunit:122

ERRORS!
Tests: 993, Assertions: 4038, Errors: 17.

Generating code coverage report in PHP format ... done [00:00.027]
make: *** [Makefile:76: test] Error 2
Error: Process completed with exit code 2.
```

--- a/application/Utils.php
+++ b/application/Utils.php
@@ -322,10 +322,12 @@ function format_date($date, $time = true, $intl = true)
         return $date->format($format);
     }
     $formatter = new IntlDateFormatter(
-        setlocale(LC_TIME, 0),
+        get_locale(LC_TIME),
         IntlDateFormatter::LONG,
         $time ? IntlDateFormatter::LONG : IntlDateFormatter::NONE
     );
+    var_dump('$formatter');
+    var_dump($formatter);
     $formatter->setTimeZone($date->getTimezone());
 
     return $formatter->format($date);
@@ -502,4 +504,24 @@ function t($text, $nText = '', $nb = 1, $domain = 'shaarli', $variables = [], $f
 function exception2text(Throwable $e): string
 {
     return $e->getMessage() . PHP_EOL . $e->getFile() . $e->getLine() . PHP_EOL . $e->getTraceAsString();
+}
+
+/**
+ * Get the current locale, overrides 'C' locale which is no longer compatible with PHP-intl
+ *
+ * @param int $category Category of the locale (LC_CTYPE, LC_NUMERIC, LC_TIME, LC_COLLATE, LC_MONETARY, LC_ALL)
+ *
+ * @return string|false The locale, or false if not found.
+ *
+ * @see https://github.com/php/php-src/issues/12561
+ */
+function get_locale(int $category = LC_CTYPE)
+{
+    $locale = setlocale($category, 0);
+
+    if ($locale === 'C' || startsWith($locale, 'C.')) {
+        $locale = 'en_US.utf8'; // failback
+    }
+
+    return $locale;
 }

--- a/application/Utils.php
+++ b/application/Utils.php
@@ -326,8 +326,6 @@ function format_date($date, $time = true, $intl = true)
         IntlDateFormatter::LONG,
         $time ? IntlDateFormatter::LONG : IntlDateFormatter::NONE
     );
-    var_dump('$formatter');
-    var_dump($formatter);
     $formatter->setTimeZone($date->getTimezone());
 
     return $formatter->format($date);

--- a/application/front/controller/visitor/FeedController.php
+++ b/application/front/controller/visitor/FeedController.php
@@ -38,7 +38,7 @@ class FeedController extends ShaarliVisitorController
         }
 
         // Generate data.
-        $this->container->feedBuilder->setLocale(strtolower(setlocale(LC_COLLATE, 0)));
+        $this->container->feedBuilder->setLocale(strtolower(get_locale(LC_COLLATE)));
         $this->container->feedBuilder->setHideDates($this->container->conf->get('privacy.hide_timestamps', false));
         $this->container->feedBuilder->setUsePermalinks(
             null !== $request->getParam('permalinks') || !$this->container->conf->get('feed.rss_permalinks')

--- a/application/http/HttpUtils.php
+++ b/application/http/HttpUtils.php
@@ -55,7 +55,7 @@ function get_http_response(
         'Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:45.0)'
         . ' Gecko/20100101 Firefox/45.0';
     $acceptLanguage =
-        substr(setlocale(LC_COLLATE, 0), 0, 2) . ',en-US;q=0.7,en;q=0.3';
+        substr(get_locale(LC_COLLATE), 0, 2) . ',en-US;q=0.7,en;q=0.3';
     $maxRedirs = 3;
 
     if (!function_exists('curl_init')) {

--- a/index.php
+++ b/index.php
@@ -81,7 +81,7 @@ if (isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
     autoLocale($_SERVER['HTTP_ACCEPT_LANGUAGE']);
 }
 
-new Languages(setlocale(LC_MESSAGES, 0), $conf);
+new Languages(get_locale(LC_MESSAGES), $conf);
 
 $conf->setEmpty('general.timezone', date_default_timezone_get());
 $conf->setEmpty('general.title', t('Shared bookmarks on ') . escape(index_url($_SERVER)));

--- a/tests/languages/de/UtilsDeTest.php
+++ b/tests/languages/de/UtilsDeTest.php
@@ -20,7 +20,7 @@ class UtilsDeTest extends UtilsTest
      */
     public function testDateFormat()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('de-de');
         $date = DateTime::createFromFormat('Ymd_His', '20170102_201112');
         $this->assertRegExp('/2\. Januar 2017 (um )?20:11:12 GMT\+0?3(:00)?/', format_date($date, true, true));
@@ -32,7 +32,7 @@ class UtilsDeTest extends UtilsTest
      */
     public function testDateFormatNoTime()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('de-de');
         $date = DateTime::createFromFormat('Ymd_His', '20170102_201112');
         $this->assertRegExp('/2\. Januar 2017/', format_date($date, false, true));
@@ -62,10 +62,10 @@ class UtilsDeTest extends UtilsTest
      */
     public function testAutoLocaleValid()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'en-us';
         autoLocale($header);
-        $this->assertEquals('en_US.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -75,10 +75,10 @@ class UtilsDeTest extends UtilsTest
      */
     public function testAutoLocaleValidAlternative()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'en_US.UTF-8';
         autoLocale($header);
-        $this->assertEquals('en_US.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -88,10 +88,10 @@ class UtilsDeTest extends UtilsTest
      */
     public function testAutoLocaleMultipleFirstValid()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'en-us,de-de';
         autoLocale($header);
-        $this->assertEquals('en_US.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -101,10 +101,10 @@ class UtilsDeTest extends UtilsTest
      */
     public function testAutoLocaleMultipleSecondAvailable()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'mgg_IN,fr-fr';
         autoLocale($header);
-        $this->assertEquals('fr_FR.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('fr_FR.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -114,9 +114,9 @@ class UtilsDeTest extends UtilsTest
      */
     public function testAutoLocaleBlank()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('');
-        $this->assertEquals('en_US.UTF-8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.UTF-8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -126,9 +126,9 @@ class UtilsDeTest extends UtilsTest
      */
     public function testAutoLocaleUnavailable()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('mgg_IN');
-        $this->assertEquals('en_US.UTF-8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.UTF-8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }

--- a/tests/languages/en/UtilsEnTest.php
+++ b/tests/languages/en/UtilsEnTest.php
@@ -20,7 +20,7 @@ class UtilsEnTest extends UtilsTest
      */
     public function testDateFormat()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('en_US.UTF-8');
         $date = DateTime::createFromFormat('Ymd_His', '20170102_201112');
         $this->assertRegExp('/January 2, 2017 (at )?8:11:12 PM GMT\+0?3(:00)?/', format_date($date, true, true));
@@ -32,7 +32,7 @@ class UtilsEnTest extends UtilsTest
      */
     public function testDateFormatNoTime()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('en_US.UTF-8');
         $date = DateTime::createFromFormat('Ymd_His', '20170102_201112');
         $this->assertRegExp('/January 2, 2017/', format_date($date, false, true));
@@ -62,10 +62,10 @@ class UtilsEnTest extends UtilsTest
      */
     public function testAutoLocaleValid()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'de-de';
         autoLocale($header);
-        $this->assertEquals('de_DE.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('de_DE.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -75,10 +75,10 @@ class UtilsEnTest extends UtilsTest
      */
     public function testAutoLocaleValidAlternative()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'de_de.UTF8';
         autoLocale($header);
-        $this->assertEquals('de_DE.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('de_DE.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -88,10 +88,10 @@ class UtilsEnTest extends UtilsTest
      */
     public function testAutoLocaleMultipleFirstValid()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'de-de;en-us';
         autoLocale($header);
-        $this->assertEquals('de_DE.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('de_DE.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -101,10 +101,10 @@ class UtilsEnTest extends UtilsTest
      */
     public function testAutoLocaleMultipleSecondAvailable()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'mgg_IN,fr-fr';
         autoLocale($header);
-        $this->assertEquals('fr_FR.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('fr_FR.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -114,9 +114,9 @@ class UtilsEnTest extends UtilsTest
      */
     public function testAutoLocaleBlank()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('');
-        $this->assertEquals('en_US.UTF-8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.UTF-8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -126,9 +126,9 @@ class UtilsEnTest extends UtilsTest
      */
     public function testAutoLocaleUnavailable()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('mgg_IN');
-        $this->assertEquals('en_US.UTF-8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.UTF-8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }

--- a/tests/languages/fr/UtilsFrTest.php
+++ b/tests/languages/fr/UtilsFrTest.php
@@ -20,7 +20,7 @@ class UtilsFrTest extends UtilsTest
      */
     public function testDateFormat()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('fr-fr');
         $date = DateTime::createFromFormat('Ymd_His', '20170102_201112');
         $this->assertRegExp('/2 janvier 2017 (à )?20:11:12 UTC\+0?3(:00)?/', format_date($date));
@@ -32,7 +32,7 @@ class UtilsFrTest extends UtilsTest
      */
     public function testDateFormatNoTime()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('fr-fr');
         $date = DateTime::createFromFormat('Ymd_His', '20170102_201112');
         $this->assertRegExp('/2 janvier 2017/', format_date($date, false, true));
@@ -62,10 +62,10 @@ class UtilsFrTest extends UtilsTest
      */
     public function testAutoLocaleValid()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'de-de';
         autoLocale($header);
-        $this->assertEquals('de_DE.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('de_DE.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -75,10 +75,10 @@ class UtilsFrTest extends UtilsTest
      */
     public function testAutoLocaleValidAlternative()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'de_de.UTF8';
         autoLocale($header);
-        $this->assertEquals('de_DE.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('de_DE.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -88,10 +88,10 @@ class UtilsFrTest extends UtilsTest
      */
     public function testAutoLocaleMultipleFirstValid()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'de-de;en-us';
         autoLocale($header);
-        $this->assertEquals('de_DE.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('de_DE.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -101,10 +101,10 @@ class UtilsFrTest extends UtilsTest
      */
     public function testAutoLocaleMultipleSecondAvailable()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         $header = 'mgg_IN,de-de';
         autoLocale($header);
-        $this->assertEquals('de_DE.utf8', setlocale(LC_ALL, 0));
+        $this->assertEquals('de_DE.utf8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -114,9 +114,9 @@ class UtilsFrTest extends UtilsTest
      */
     public function testAutoLocaleBlank()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('');
-        $this->assertEquals('en_US.UTF-8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.UTF-8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }
@@ -126,9 +126,9 @@ class UtilsFrTest extends UtilsTest
      */
     public function testAutoLocaleUnavailable()
     {
-        $current = setlocale(LC_ALL, 0);
+        $current = get_locale(LC_ALL);
         autoLocale('mgg_IN');
-        $this->assertEquals('en_US.UTF-8', setlocale(LC_ALL, 0));
+        $this->assertEquals('en_US.UTF-8', get_locale(LC_ALL));
 
         setlocale(LC_ALL, $current);
     }

